test/should_fail/covariant_map_keys_fail.erl

```diff
@@ -3,11 +3,19 @@
 -compile([nowarn_unused_function]).
 -export([not_good/1]).
 
--spec good(#{ good := A }) -> A.
-good(#{ good := X }) -> X.
+-gradualizer([infer]).
```

Comment:
The test suite is a little disorganized wrt `infer`. But I think it would be helpful to have separate files for things which are tested with `infer` enabled and have `infer` in the name.

Reply:
Good point, I'll split out the parts relying on `infer`.